src/test/java/org/gaul/s3proxy/junit/S3ProxyRuleTest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2014-2018 Andrew Gaul <andrew@gaul.org>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gaul.s3proxy.junit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.client.builder.AwsClientBuilder.EndpointConfiguration;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.Bucket;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+
+import com.google.common.io.Files;
+
+import org.gaul.s3proxy.junit.S3ProxyRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+/**
+ * This is an example of how one would use the S3Proxy JUnit rule in a unit
+ * test as opposed to a proper test of the S3ProxyRule class.
+ */
+public class S3ProxyRuleTest {
+
+    private static final String MY_TEST_BUCKET = "my-test-bucket";
+
+    private static final String AWS_ACCESS_KEY = "access";
+    private static final String AWS_SECRET_KEY = "secret";
```

Comment:
You don't actually need these fields since you call `S3ProxyRule.getAccessKey` below.

Reply:
Left over from our own test which this code originally came from. Have inlined these as suggested.